File: snail-race-bet/typescript/src/domain/BetBuilder.ts
import {Bet, PodiumPronostic} from "./BetRepository";


export class BetBuilder {
    private timestamp: number = Math.round(Math.random() * 1000);

    public withTimeStamp(timestamp: number): BetBuilder {
        this.timestamp = timestamp;
        return this;
    }

    public build(): Bet {
        return new Bet("mathieu", new PodiumPronostic(4, 5, 6), this.timestamp);
    }
}

File: snail-race-bet/java/src/test/java/snail/race/kata/domain/BetBuilder.java
package snail.race.kata.domain;

public class BetBuilder {

    private int timestamp = (int)Math.round(Math.random()*1000);

    public BetBuilder withTimeStamp(int timpestamp) {
        this.timestamp = timpestamp;
        return this;
    }

    public Bet build() {
        return new Bet("mathieu", new PodiumPronostic(4,5,6), timestamp);
    }
}
